src/status_im/contexts/chat/messenger/messages/list/view.cljs

```diff
@@ -359,17 +169,15 @@
         (reset! content-height content-height-new)))))
 
 (defn on-layout
-  [{:keys [event layout-height distance-atom distance-from-list-top
-           chat-screen-layout-calculations-complete?]}]
+  [{:keys [event layout-height distance-atom distance-from-list-top on-layout-done?]}]
   (let [layout-height-new (oops/oget event "nativeEvent.layout.height")
         change            (- layout-height-new @layout-height)
         new-distance      (- @distance-atom change)]
     (when (and (not= change 0) (not= @layout-height layout-height-new))
       (reanimated/set-shared-value distance-from-list-top new-distance)
       (reset! distance-atom new-distance)
       (reset! layout-height layout-height-new))
-    (when-not (reanimated/get-shared-value chat-screen-layout-calculations-complete?)
-      (reanimated/set-shared-value chat-screen-layout-calculations-complete? true))))
```

Comment:
The layout calculation is just a boolean marked once the `on-layout` event was fired. It's been simplified to only be a ratom switch.